circom/circom_test_harness.py

```diff
@@ -0,0 +1,137 @@
+import json
+import sys
+import os.path
+import subprocess
+from argparse import ArgumentParser
+
+class CircomComponent():
+    def __init__(self, name: str, lib: str) -> None:
+        self.name = name
+        self.lib = lib
+
+class CircomComponentInstance(CircomComponent):
+    def __init__(self, name: str, lib: str, params: list[int]) -> None:
+        super().__init__(name, lib)
+        self.params = params
+    
+    def get_runner(self) -> str:
+        return self.lib + '.' + self.name + '_' + '_'.join(map(str, self.params)) + '_runner'
```

Comment:
```suggestion
        return f'{self.lib}.{self.name}_{"_".join(map(str, self.params))}_runner'
```
I think f-strings look more readable; applicable to other lines as well.